src/redis-cli.c

```diff
@@ -3247,16 +3247,20 @@ void cliLoadPreferences(void) {
 /* Some commands can include sensitive information and shouldn't be put in the
  * history file. Currently these commands are include:
  * - AUTH
- * - ACL SETUSER
+ * - ACL SETUSER, ACL GETUSER, ACL DRYRUN
  * - CONFIG SET masterauth/masteruser/requirepass
  * - HELLO with [AUTH username password]
- * - MIGRATE with [AUTH password] or [AUTH2 username password] */
+ * - MIGRATE with [AUTH password] or [AUTH2 username password] 
+ * - SENTINEL CONFIG SET sentinel-pass password, SENTINEL CONFIG SET sentinel-user username 
+ * - SENTINEL SET <mastername> auth-pass password, SENTINEL SET <mastername> auth-user username */
 static int isSensitiveCommand(int argc, char **argv) {
     if (!strcasecmp(argv[0],"auth")) {
         return 1;
     } else if (argc > 1 &&
-        !strcasecmp(argv[0],"acl") &&
-        !strcasecmp(argv[1],"setuser"))
+        !strcasecmp(argv[0],"acl") && (
+            !strcasecmp(argv[1],"setuser") ||
+            !strcasecmp(argv[1],"getuser") ||
+            !strcasecmp(argv[1],"dryrun")))
```

Comment:
why is GETUSER sensitive?
also, why is DRYRUN? in theory it can be used to run AUTH, but it is more likely it'll be used for normal commands such as SET, PUBLISH, FLUSHDB

Reply:
Because I find the command ACL SETUSER username [rule [rule ...]] and CONFIG SET masteruser username, both of these commands have the username, and they are sensitive commands, Thus I think the commands ACL DRYRUN username command [arg [arg ...]] and ACL GETUSER username should be sensitive commands as well.

Reply:
ohh ok, because they have the user name... would be nice to add a comment.

i think that for DRYRUN, not to include it in the history would have a very high impact.
usually, when someone types s DRYRUN, he'll repeat it with different variations several times, and also unlike GETUSER, it is likely to be a long line.

i'd like to take this opportunity to consider two alternatives.
1. keep an in-memory history of all the redacted commands, and just not persist it. this way you would be able to press the up arrow and re-try the command freely, and it'll just not survive a redis-cli restart. (i'm not sure if it's acceptable, and what other similar tools do).
2. keep logging the command, and just redact the sensitive arguments.

@yossigo @madolson WDYT?

Reply:
I like the in-memory copy of sensitive commands. I do periodically run into the problem where I can't repeat sensitive commands while testing, it would be nice if it was kept in memory.

Reply:
@hwware i'll merge this PR, but after it, maybe you can make a PR to implement that idea of keeping full history in memory, and just avoiding storing sensitive commands in the persisted history file?